WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -0,0 +1,185 @@
+        case .WordPressCom:
+            return wpcomPreferenceKeys
+        }
+    }
+    
+
+    /**
+    *  @enum        Channel
+    *  @brief       Represents a communication channel that may post notifications to the user.
+    */
+    public enum Channel : Equatable {
+        case Blog(blogId: Int)
+        case Other
+        case WordPressCom
+        
+        
+        /**
+        *  @details Returns the localized description of the current enum value
+        */
+        func description() -> String {
+            switch self {
+            case .Blog:
+                return NSLocalizedString("WordPress Blog", comment: "Settings for a Wordpress Blog")
+            case .Other:
+                return NSLocalizedString("Comments on Other Sites", comment: "Notification Settings Channel")
+            case .WordPressCom:
+                return NSLocalizedString("Updates from WordPress.com", comment: "Notification Settings Channel")
+            }
+        }
+    }
+    
+    
+    /**
+    *  @class       Stream
+    *  @brief       Contains the Notification Settings collection for a specific communications stream.
+    */
+    public class Stream {
+        public var kind         : Kind
+        public var preferences  : [String : Bool]?
+        
+        
+        /**
+        *  @details Designated Initializer
+        *  @param   kind            The Kind of stream we're currently dealing with
+        *  @param   preferences     Raw remote preferences, retrieved from the backend
+        */
+        public init(kind: String, preferences: [String : Bool]?) {
+            self.kind           = Kind(rawValue: kind) ?? .Email
+            self.preferences    = preferences
+        }
+        
+        
+        /**
+        *  @enum    Stream.Kind
+        *  @brief   Enumerates all of the possible Stream Kinds
+        */
+        public enum Kind : String {
+            case Timeline       = "timeline"
+            case Email          = "email"
+            case Device         = "device"
+            
+            
+            /**
+            *  @details Returns the localized description of the current enum value
+            */
+            func description() -> String {
+                switch self {
+                case .Timeline:
+                    return NSLocalizedString("Notifications Tab", comment: "WordPress.com Notifications Timeline")
+                case .Email:
+                    return NSLocalizedString("Email", comment: "Email Notifications Channel")
+                case .Device:
+                    return NSLocalizedString("Push Notifications", comment: "Mobile Push Notifications")
+                }
+            }
+            
+            static let allValues = [ Timeline, Email, Device ]
+        }
+    }
+    
+    // MARK: - Private Properties
+    private let blogPreferenceKeys      = ["new_comment", "comment_like", "post_like", "follow", "achievement", "mentions"]
```

Comment:
The consumers actually retrieve the list of keys they need to render (and their descriptions). **However** there's duplication within this file itself, which is error prone.

Meaning that you're definitely right! Updating this one as well, thanks *a lot* for spotting this!